airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java

```diff
@@ -356,6 +356,14 @@ protected void initTests() {
             .addExpectedValues("99999", "5.1", "0", null)
             .build());
 
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .fullSourceDataType("numeric(13,4)")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("0.1880", "null")
+            .addExpectedValues("0.1880", null)
+            .build());
+
```

Comment:
Might be worth adding in some more values to be more secure here e.g.:
.addInsertValues("0.1880", "{another value}", "{another value}", "{another value"}, "null")
 .addExpectedValues("0.1880", "{another value}", "{another value}", "{another value"}, null)

(perhaps use ones from your example problem that were syncing incorrectly)